Observer/AdyenHppDataAssignObserver.php

```diff
@@ -118,12 +123,17 @@ public function execute(Observer $observer)
         if (!empty($stateData)) {
             $stateData = $this->checkoutStateDataValidator->getValidatedAdditionalData($stateData);
         }
+
+        if ($additionalData[self::BRAND_CODE] === Data::SEPA) {
```

Comment:
Hi

Currently I'm working on GraphQL implementation Adyen plugin with our frontend, so I pulled develop and everything was working well, but after these update and adding these code

if ($additionalData[self::BRAND_CODE] === Data::SEPA) { $additionalDataToSave = $this->getSepaAdditionalDataToSave($stateData); }

into the AdyenHppDataAssignObserver it stopped working because after second time when this event is triggering the additional data received from payment object (table) there is no brand_code field because in the payment table no such field, this data stored in additional_information field so as the temporary solution we can add these code

if (isset($additionalData['additional_information'])) { $additionalData = array_merge($additionalData, $additionalData['additional_information']); }

right above the previous the

$additionalData = DataArrayValidator::getArrayOnlyWithApprovedKeys( $additionalData, self::$approvedAdditionalDataKeys );

to add brand_code from additional_information

this error can be reproduced within HPP method as PP for example

UPD

or we could check the cc_type field instead of brand_code because in case of HPP it stored brand code value, for example for PP cc_type = paypal

Reply:
Hi @evgv,

Thanks for your comment. Just to check that we are on the same page, you're saying that a simple solution would be to check the array before accessing the `self::BRAND_CODE` right? So something like:

```
if (array_key_exists(self::BRAND_CODE, $additionalData) && $additionalData[self::BRAND_CODE] === Data::SEPA) {
```

Thanks,
Jean
Adyen

Reply:
Yep, you are right!

Reply:
Check added in https://github.com/Adyen/adyen-magento2/pull/1395/commits/102c0725267ed7f0e849078cae9769c96c46f94f